compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -672,8 +672,10 @@ pub enum TerminatorKind<'tcx> {
     /// necessarily executed even in the case of a panic, for example in `-C panic=abort`. If the
     /// assertion does not fail, execution continues at the specified basic block.
     ///
-    /// When overflow checking is disabled and we are generating run-time code, the `Overflow*`
-    /// variants of this terminator are codegened as simple `goto target`.
+    /// When overflow checking is disabled and this is run-time MIR (as opposed to compile-time MIR
+    /// that is used for CTFE), the following variants of this terminator behave as `goto target`:
+    /// - `OverflowNeg(..)`,
+    /// - `Overflow(op, ..)` if op is a checkable operation (add, sub, mul, shl, shr).
```

Comment:
Why not just do this for all `op`? That seems simpler?

Reply:
There is the case of overflowing division. Currently, the overflow check is always emitted for `BinOp::Div` and `BinOp::Rem`. AFAICT, it's the only exception here. None of the other operations can overflow.

Reply:
Ah, good point. I have long wondered about this inconsistency.

Reply:
The term "checkable" is a strange one though since Div/Rem are also checkable... I realize `is_checkable` predates this PR but the name still makes no sense to me.

```suggestion
    /// - `Overflow(op, ..)` if op is a "checkable" operation (add, sub, mul, shl, shr, but NOT
    /// div or rem).
```